### DISCUSSION TITLE
constraint Jacobian mis-allocation

### DISCUSSION THREAD

**Discussion by nisal-perera** [google-deepmind/mujoco#2187] [Asking for Help] [answered]
### Intro

Hi!

I am a PhD student using Mujoco to simulate a bipedal robot with closed chain mechanism. 

### My setup

I am using Mujoco 3.2.4 on ubuntu 20.04 with C++.

### My question

I made a mujoco model for my biped robot and created the closed-chain mechanism using equality (<connect>) constraint. I made sure to use the bare minimum number of equality constraints to defined my model. When i fix the robot w.r.t. the world (i.e. remove the <free joint>) i can simulate the robot with the linkages. However, when i add the <free joint> to make the robot a floating base, I get cannot simulate the model. I get the following error **"Error:engine error: mj_makeConstraint: constraint Jacobian mis-allocation:found nnzJ=642 but allocated 594"**.

I tried increasing the njmax and nconmax and other size values but it did not seem to help fixing this issue. Any help to get around this issue is appreciated.

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
This is a serious bug! That error message should never happen.

Please submit a proper bug report with a minimal reproducible example and we will look into it ASAP.

Please take care to make your MRE
1. **Minimal**: try to reduce the model as much as possible while still maintaining the error. Specifically if you could make a model that does not require binary assets (meshes/textures), that would be highly appreciated.
2. **Reproducible**: Make sure we have everything we need to reproduce the error.

**Reply by aryakeyvan-star** [reply]
I am getting the same error message. Are there any work arounds to get around this issue?

**Reply by yuvaltassa** [reply, marked as answer]
This is fixed since 3.2.5

**Reply by Wan020607** [reply]
Will the above problem happen in issac sim if I want to use mjcf
